src/report/ReportPoller.ts

```diff
@@ -104,18 +110,29 @@ export class ReportPoller {
             });
         }
 
-        /*
-         * This API endpoint returns an opaque `next_token` number that we
-         * need to give back to subsequent requests for pagination, so here we
-         * save it in account data
-         */
-        if (response.next_token !== undefined) {
-            this.from = response.next_token;
-            try {
-                await this.mjolnir.client.setAccountData(REPORT_POLL_EVENT_TYPE, { from: response.next_token });
-            } catch (ex) {
-                await this.mjolnir.logMessage(LogLevel.ERROR, "getAbuseReports", `failed to update progress: ${ex}`);
-            }
+        let from;
+        if (this.from === null) {
+            /*
+             * If this is our first call to this endpoint, we want to skip to
+             * the end of available reports, so we'll only consider reports
+             * that happened after we supported report polling.
+             */
+            from = response.total;
+        } else {
+            /*
+             * If there are more pages for us to read, this endpoint will
+             * return an opaque `next_token` number that we want to provide
+             * on the next endpoint call. If not, we're on the last page,
+             * which means we want to skip to the end of this page.
+             */
+            from = response.next_token ?? this.from + response.event_reports.length;
```

Comment:
```ts
this.from + response.event_reports.length
```
isn't ideal because the `from` token is meant to be opaque, but i don't see a better way to do this. i've been in discussions with synapse devs to talk about how this endpoint could be better